Lib/defconQt/glyphView.py

```diff
@@ -1651,62 +2138,52 @@ def createAnchor(self):
             anchor.name = newAnchorName
             self._glyph.appendAnchor(anchor)
 
-    def createComponent(self):
+    def createComponent(self, *args):
         newGlyph, ok = AddComponentDialog.getNewGlyph(self, self._glyph)
         if ok and newGlyph is not None:
             component = Component()
             component.baseGlyph = newGlyph.name
             self._glyph.appendComponent(component)
 
-    def _currentLayerChanged(self, newLayerIndex):
-        comboBox = self.sender()
-        newLayer = comboBox.itemData(newLayerIndex)
-        if newLayer is None:
-            # add a new layer
-            newLayerName, ok = AddLayerDialog.getNewLayerName(self)
-            if ok:
-                self._glyph.layerSet.newLayer(newLayerName)
-                self._currentLayerBox.blockSignals(True)
-                self._currentLayerBox.setCurrentText(newLayerName)
-                self._currentLayerBox.blockSignals(False)
-                newLayer = self._glyph.layerSet[newLayerName]
-            else:
-                return
-        if not self._glyph.name in newLayer:
-            newLayer.newGlyph(self._glyph.name)
-            # TODO: generalize this out, can’t use newStandardGlyph unfortunately
-            newLayer[self._glyph.name].width = self._glyph.width
-            newLayer[self._glyph.name].template = True
-        newGlyph = newLayer[self._glyph.name]
-        self.setGlyph(newGlyph)
-
-    def setGlyph(self, glyph):
+    def _makeLayerGlyph(self, layer):
+        name = self._name
+        layer.newGlyph(name)
+        glyph = layer[name]
+        # TODO: generalize this out, can’t use newStandardGlyph unfortunately
+        glyph.width = self.defaultWidth
+
+        # What does this do actually?
```

Comment:
Hm, I think it's because the save exclusion mechanism wasn't upgraded to the ufo3 method as in https://github.com/typesupply/defcon/issues/24#issuecomment-144384072